src/lib.rs

```diff
@@ -38,6 +37,34 @@ fn sort_lines(lines: Vec<String>) -> Vec<String> {
     lines
 }
 
+fn streaming_sort_until<'a, IN, OUT, STR>(
+    input: IN,
+    mut output: OUT,
+    end: &str,
+) -> std::io::Result<()>
+where
+    // can take &str or &String
+    STR: AsRef<str>,
+    IN: Iterator<Item = STR>,
+    OUT: Write,
+{
+    let input = input
+        .take_while(|l| l.as_ref() != end)
+        .map(|l| TsvLine::new(l.as_ref()));
+    // Do the external sort
+    let sorted = ExternalSorter::new(1000000, None)
+        .sort_by(input, |a, b| {
+            tsv_cmp(a.the_line.as_str(), b.the_line.as_str())
+        })
+        .unwrap();
```

Comment:
Forgive my ignorance, but could this `.unwrap()` be a `?`?

Reply:
It requires a different error type in the signature of this function, one that can be constructed from the ExternalSorter error. Maybe PyError can be constructed from arbitrary errors? Requires investigation